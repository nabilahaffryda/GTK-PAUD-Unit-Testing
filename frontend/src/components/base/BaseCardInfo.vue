<template>
  <v-card flat outlined class="my-2">
    <v-list-item>
      <v-list-item-avatar :size="size" color="primary">
        <v-icon v-if="icon" dark v-text="icon"></v-icon>
      </v-list-item-avatar>
      <v-list-item-content>
        <v-list-item-title v-html="title"></v-list-item-title>
      </v-list-item-content>
    </v-list-item>
    <v-card-text class="pt-0">
      <span class="ml-12 headline blue--text" v-html="info"></span>
      <slot></slot>
    </v-card-text>
  </v-card>
</template>

<script>
export default {
  name: 'CardInfo',
  props: {
    size: {
      type: Number,
      default: 28,
    },
    icon: {
      type: String,
      default: '',
    },
    title: {
      type: String,
      default: '',
    },
    info: {
      type: String,
      default: '',
    },
  },
};
</script>

<style scoped></style>
