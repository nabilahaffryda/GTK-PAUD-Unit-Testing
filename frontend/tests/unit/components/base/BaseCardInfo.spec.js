import { mount, createLocalVue } from "@vue/test-utils";
import BaseCardInfo from "@/components/base/BaseCardInfo.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';

Vue.use(Vuetify)
const localVue = createLocalVue();

describe('BaseCardInfo.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })
    test('should show card info', () => {
        const VCard = {
            props: ['size', 'icon', 'title', 'info'],
            template: '<div><slot :size="size" /></div>'
        }

        const wrapper = mount(BaseCardInfo, {
            localVue,
            vuetify,
            stubs: {
                VCard
            }
        });
        expect(wrapper).toMatchSnapshot();
    })
})
