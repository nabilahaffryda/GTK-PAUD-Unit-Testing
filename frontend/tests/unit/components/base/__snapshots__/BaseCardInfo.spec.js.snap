// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseCardInfo.vue should show card info 1`] = `
<div flat="" outlined="" class="my-2">
  <div tabindex="-1" class="v-list-item theme--light">
    <div class="v-avatar v-list-item__avatar primary" style="height: 28px; min-width: 28px; width: 28px;"></div>
    <div class="v-list-item__content">
      <div class="v-list-item__title"></div>
    </div>
  </div>
  <div class="v-card__text pt-0"><span class="ml-12 headline blue--text"></span> </div>
</div>
`;
